src/inmanta/compiler/__init__.py

```diff
@@ -337,22 +338,28 @@ def call_finalizers(cls, should_log: bool = False) -> None:
         by default this function will raise exceptions caused by errors in the finalizer functions
         if 'should_log' is set to True the exceptions will not be raised but logged instead.
         """
-        try:
-            for fnc in cls.__finalizers:
+        excns: list[CompilerException] = []
+        for fnc in cls.__finalizers:
+            try:
                 fnc()
-        except Exception as e:
+            except Exception as e:
+                excns.append(CompilerException("Finalizer failed: " + str(e)))
+        if excns:
             if should_log:
-                LOGGER.error(f"Finalizers failed: {e}")
+                msg: list[str] = []
+                for exception in excns:
+                    msg.append(exception.msg + "\n")
+                LOGGER.error("".join(msg).strip())
```

Comment:
Since these are separate failures I have a slight preference for making it separate log statements. Not a strict change requirement.